### ISSUE TITLE
clippy::ptr_arg suggests changing &mut Vec to &mut [u8] even when a method only applicable to Vec is applied.

### ISSUE THREAD

**Issue by John-Nagle** [rust-lang/rust-clippy#9642]
### Summary

Following Clippy's suggestion results in a program that will not compile.

Clippy wants an argument changed from **&mut Vec** to **&mut [u8]**. But there's a macro call applied to that argument which applies **Vec::extend_from_slice** to it. Clippy did not notice that the macro contained something that made this change invalid.

### Lint Name

#[warn(clippy::ptr_arg)]

### Reproducer

I tried this code:

```rust
<code>
```

I saw this happen:

```
warning: writing `&mut Vec` instead of `&mut [_]` involves a new object where a slice will do
    --> libclient/src/messages/msgvolumeextras.rs:1426:13
     |
1426 |         _b: &mut Vec<u8>,
     |             ^^^^^^^^^^^^ help: change this to: `&mut [u8]`


error[E0599]: no method named `extend_from_slice` found for mutable reference `&mut [u8]` in the current scope
   --> libclient/src/messages/helpers.rs:178:14
    |
178 |           $buf.extend_from_slice(&<$typ>::to_le_bytes($value)); // push onto output vector
    |                ^^^^^^^^^^^^^^^^^ help: there is an associated function with a similar name: `clone_from_slice`


```

I expected to see this happen:

### Version

```text
rustc -Vv
rustc 1.62.1 (e092d0b6b 2022-07-16)
binary: rustc
commit-hash: e092d0b6b43f2de967af0887873151bb1c0b18d3
commit-date: 2022-07-16
host: x86_64-unknown-linux-gnu
release: 1.62.1
LLVM version: 14.0.5
```

**Comment by Jarcho** [comment]
Can you post the whole code for the function in question? This is not enough to reproduce the issue.

**Comment by John-Nagle** [comment]
My error. Sorry. In trying to fix a Clippy warning, I changed the wrong &mut Vec<u8>, to &mut [u8].

Clippy had properly followed the parameter into the macro that uses .extends_from_slice and detected that it really needed to be a Vec.

**Comment by John-Nagle** [comment]
Yes, close. My bad.
